archive_repo.py

```diff
@@ -0,0 +1,667 @@
+  }
+}
+"""
+    + gql_Issue_Fields
+)
+
+gql_AllIssues_First = (
+    """
+query($owner: String!, $repo: String!){
+  repository(owner: $owner, name: $repo) {
+    issues(first: 100) {
+"""
+    + gql_Issues_Query
+)
+
+gql_AllIssues_Subsequent = (
+    """
+query($owner: String!, $repo: String!, $cursor: String!){
+  repository(owner: $owner, name: $repo) {
+    issues(first: 100, after: $cursor) {
+"""
+    + gql_Issues_Query
+)
+
+gql_UpdatedIssues_First = (
+    """
+query($owner: String!, $repo: String!, $filters: IssueFilters){
+  repository(owner: $owner, name: $repo) {
+    issues(first: 100, filterBy: $filters) {
+"""
+    + gql_Issues_Query
+)
+
+gql_UpdatedIssues_Subsequent = (
+    """
+query($owner: String!, $repo: String!, $filters: IssueFilters, $cursor: String!){
+  repository(owner: $owner, name: $repo) {
+    issues(first: 100, filterBy: $filters, after: $cursor) {
+"""
+    + gql_Issues_Query
+)
+
+gql_Issue_Comments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on Issue {
+            comments(first:100, after:$cursor) {
+                nodes {
+                    ...author
+                    ...commentFields
+                }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+    + gql_AuthorFields
+)
+
+# Pull Requests
+
+gql_Review_Fields = (
+    """
+fragment reviewFields on PullRequestReview {
+    id
+    commit { abbreviatedOid }
+    ...author
+    state
+    ...commentFields
+    comments(first: 50) {
+        nodes {
+            originalPosition
+            ...commentFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_Comment_Fields
+    + gql_AuthorFields
+)
+
+gql_PullRequest_Fields = (
+    """
+fragment prFields on PullRequest {
+    number
+    id
+    title
+    url
+    state
+    ...labels
+    ...author
+    ...commentFields
+    closedAt
+    mergedAt
+    mergedBy { ...userFields }
+    comments(first: 100) {
+        nodes {
+            ...author
+            ...commentFields }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+    reviews(first: 50) {
+        nodes {
+            ...reviewFields
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+        }
+        totalCount
+    }
+}
+"""
+    + gql_LabelFields
+    + gql_Review_Fields
+)
+# ...reviewFields definition includes ...commentFields and ...author
+# ...author definition includes ...userFields
+
+gql_PullRequest_Query = (
+    """
+      nodes { ...prFields }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+      totalCount
+    }
+  }
+  rateLimit {
+    remaining
+    resetAt
+  }
+}
```

Comment:
Unbalanced braces do make this slightly harder to read.

Reply:
The alternative is to repeat this piece several times.  I figured proper indentation could make this the lesser of two evils.

Reply:
It's just one line.

Reply:
Then I think we're talking about different things.  Do you have a suggested change?